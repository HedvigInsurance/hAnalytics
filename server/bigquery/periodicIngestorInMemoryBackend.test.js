const inMemoryBackend = require("./periodicIngestorInMemoryBackend");

test("stores and consumes correctly", async () => {
  inMemoryBackend.append({
    mock: "mock",
  });

  expect(await inMemoryBackend.consume()).toMatchSnapshot();
  expect(await inMemoryBackend.consume()).toMatchSnapshot();

  [...new Array(5000)].forEach(() => {
    inMemoryBackend.append({
      mock: "mock",
    });
  });

  expect((await inMemoryBackend.consume()).length).toEqual(5000);
  expect((await inMemoryBackend.consume()).length).toEqual(0);
});
