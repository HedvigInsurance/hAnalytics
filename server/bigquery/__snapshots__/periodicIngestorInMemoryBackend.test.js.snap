// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stores and consumes correctly 1`] = `
Array [
  Object {
    "mock": "mock",
  },
]
`;

exports[`stores and consumes correctly 2`] = `Array []`;
